### ISSUE TITLE
One-click spam button

### ISSUE THREAD

**Issue by timgl** [papercups-io/papercups#968]
**Problem**
We get quite a lot of unsollicited/spam emails. 

**Solution**
Would love to have a 1click in slack to mark as spam, delete the message from Slack (or at least collapse it as those spam emails tend to take up a lot of channel real estate) and  never let emails from that sender through again.

**Describe alternatives you've considered**
I manually 'mark as resolved', then delete the slack message itself so it doesn't distract anyone else. Usually however these people will follow up so I need to repeat the song and dance.

**Additional context**
Add any other context, screenshots, or related issues about the feature request here.

**Testing**
Describe how this should be tested.

**Questions, or need help getting started?**
Feel free to ask below, or [ping us on Slack](https://join.slack.com/t/papercups-io/shared_invite/zt-h0c3fxmd-hZi1Zp8~D61S6GD16aMqmg) :)

(You can also check out our [CONTRIBUTING.md](https://github.com/papercups-io/papercups/blob/master/CONTRIBUTING.md))

**Comment by reichert621** [comment]
good idea @timgl — would you also want these marked as spam within Gmail (e.g. we could label it as spam via the API), or just internally within Papercups?

**Comment by timgl** [comment]
I think marking them as a spam in gmail is a nice to have, doesn't really affect us that much (except for punishing spammers which is always good :))

**Comment by timgl** [comment]
Any other thoughts on this? A decent chunk of the messages we have to wade through now are spam messages and makes us lose track of real messages in Slack so would be awesome to have this

**Comment by reichert621** [comment]
@timgl as a first pass, we could surface a "delete" button in slack that handles deleting the thread both in slack and in the dashboard -- would that be ok for now?

**Comment by timgl** [comment]
Yep that'd be great!
